EIPS/eip-7637.md

```diff
@@ -0,0 +1,94 @@
+---
+eip: 7637
+title: EXTCODEHASH optimize
+description: Modify the output value of a situation in EXTCODEHASH
+author: Jame (@ZWJKFLC)
+discussions-to: https://ethereum-magicians.org/t/eip-7637-extcodehash-optimize/18946
+status: Draft
+type: Standards Track
+category: Core
+created: 2024-02-26
+---
+
+
+
+## Abstract
+
+This EIP proposal is an optimization for [EIP-1052](./eip-1052.md),
+For addresses with balance but the code is still 0x, the codehash should still be 0x.
+
+
+## Motivation
+
+`EIP-1052` was proposed to save gas fees.
+In order to include the role of `BALANCE`, let the codehash of the address without balance be 0x, and the codehash of the address with balance be hash(0x).
```

Comment:
```suggestion
In order to include the role of `BALANCE`, let the codehash of the address without balance be `0x`, and the codehash of the address with balance be `hash(0x)`.
```
I'm not sure what this sentence is trying to say.

Reply:
This is the definition of eip-1052.
He wants to be compatible with the use of `BALANCE`.
When `add.code == 0x`,
If an address `add.balance == 0` then `add.codehash==0`.
If an address `add.balance != 0` then `add.codehash==keccak256(bytes(""))`
This makes it unreliable to use codehash to determine whether a contract is created, which defeats its original purpose of saving gas.